File: productHuntServer/src/main/java/com/nagarro/service/BrandService.java
package com.nagarro.service;

import java.util.Set;

import com.nagarro.model.product.Brand;

public interface BrandService {
	
	public Brand addBrand(Brand brand);
	public Brand updateBrand(Brand brand);
	public Set<Brand> getBrands();
	public Brand getBrand(Long brandId);
	public void deleteBrand(Long brandId);
	public long countAllBrands();
}

File: productHuntFrontend/src/app/services/brand.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root',
})
export class BrandService {
  constructor(private _http: HttpClient) {}

  //load all the brands

  public brands() {
    return this._http.get(`${baseUrl}/brand/`);
  }

  //add new brand
  public addBrand(brand: any) {
    return this._http.post(`${baseUrl}/brand/`, brand);
  }

  //delete brand
  public deleteBrand(bId: any) {
    return this._http.delete(`${baseUrl}/brand/${bId}`);
  }

  //get the single brand
  public getBrand(bId: any) {
    return this._http.get(`${baseUrl}/brand/${bId}`);
  }

  //update brand
  public updateBrand(brand: any) {
    return this._http.put(`${baseUrl}/brand/`, brand);
  }

  //total brands
  public countOfBrands(): Observable<any> {
    return this._http.get(`${baseUrl}/brand/count`);
  }
}
